src/System.IO.Pipelines/ReadCursor.cs

```diff
@@ -157,7 +130,7 @@ private ReadCursor SeekMultiSegment(int bytes, ReadCursor end, bool checkEndReac
                 {
                     if (segmentPart.Length >= bytes)
                     {
-                        result = new ReadCursor(segmentPart.Segment, segmentPart.Start + bytes);
+                        result = new ReadCursor(segmentPart.Segment, segmentPart.Start + (int)bytes);
```

Comment:
`ReadCursor` stores absolute position inside segment. And it's not `=> segmentPart.End` it's maximum value is `segmentPart.End` which is supported.